src/components/forums/forum/ForumContent.tsx

```diff
@@ -43,15 +46,96 @@ import {
 } from '../../../utils/restrictionListHelper';
 import { newPublicKey } from '../../../utils/postbox/validateNewPublicKey';
 import { csvStringToPubkeyList } from '../../../utils/csvStringToPubkeyList';
-import { StarsAlert } from '../StarsAlert';
 
 interface ForumContentProps {
   forumObject: DispatchForum;
-  forumData: ForumData;
+  basicInfo?: { title: string; desc: string };
+  forumData?: ForumData;
   update: () => Promise<void>;
 }
 
 export function ForumContent(props: ForumContentProps): JSX.Element {
+  const { forumData, forumObject, basicInfo, update } = props;
+  const { permission } = forumObject;
+
+  if (isNil(forumData)) {
+    const confirmingBox = (
+      <div className='confirmingBanner'>
+        <div className='title'>
+          <div className='animation'>
+            <Lottie
+              loop
+              animationData={animationData}
+            />
+          </div>
+          <div className='text'>The network is confirming your forum.</div>
+        </div>
+        <div className='subtitle'>When it`s ready, the page will reload itself. This may take a few seconds.</div>
+      </div>
+    );
+
+    if (!isNil(basicInfo)) {
+      const forumHeader = (
+      <div className="forumContentHeader">
+        <div className={'titleBox'}>
+          <Markdown>{basicInfo.title}</Markdown>
+        </div>
+        <div className="descriptionBox">
+          <div className="description">
+            <Markdown>{basicInfo.desc}</Markdown>
+          </div>
+          <button
+            className={'createTopicButton'}
+            type="button"
+            disabled
+            >
+            <div className="buttonImageContainer">
+              <Plus />
+            </div>
+            Create Topic
+          </button>
+        </div>
+      </div>
+      );
+
+      return (
+      <div className='confirmingWrapper'>
+        <div className="forumContent">
+          <>{ReactGA.send('pageview')}</>
+          <div className="forumContentBox" >
+            {!permission.readAndWrite && <ConnectionAlert />}
+            {confirmingBox}
+            {forumHeader}
+          </div>
+          <div className="toolsWrapper" />
+          <div className="topicListWrapper">
+            <TopicList />
+          </div>
+        </div>
+      </div>
+      );
+    } else {
+      return <Spinner/>;
+    }
+  } else {
+    return (
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      <PopulatedForumContent
+        forumObject={forumObject}
+        forumData={forumData}
+        update={update}
+      />
+    );
+  }
+}
+
+interface PopulatedForumContentProps {
+  forumObject: DispatchForum;
+  forumData: ForumData;
+  update: () => Promise<void>;
+}
+
+export function PopulatedForumContent(props: PopulatedForumContentProps): JSX.Element {
```

Comment:
we should avoid having two components in the same file, can you move this to a dedicated file?

Reply:
I agree with avoiding that, but in this case, the idea is to have one component that acts as an intermediary between the populated version of the component and the confirming one. It doesn't make sense to separate these two in different files since they're basically the same one, just in different phases (they even share classnames)

Reply:
that makes sense, i think it works for now then